File: arb-www/src/schema/rollingInsights/nba/playerProfile.ts
import { BaseDTO } from '../../BaseDTO';

/**
 * NBA Rolling Insights Player Profile
 */
export class NBARollingInsightsPlayerProfile extends BaseDTO {
  player_id!: number;
  player!: string;
  team_id!: number;
  team!: string;
  number!: number;
  status!: string;
  position!: string;
  position_category!: string;
  height!: string;
  weight!: number;
  age!: string;
  college!: string;
}

/**
 * Rolling Insights NBA Player Profile response wrapper
 */
export interface NBARollingInsightsPlayerProfileResponse {
  league: string;
  data_type: string;
  data: {
    Nba: {
      PlayerProfiles: NBARollingInsightsPlayerProfile[];
    };
  };
  filtered_count: number;
  total_count: number;
}

File: arb-rs/src/schema/rolling_insights/nba/player_profile.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NBARollingInsightsPlayerProfile {
    #[serde(rename = "player_id")]
    pub player_id: i32,
    pub player: String,
    #[serde(rename = "team_id")]
    pub team_id: i32,
    pub team: String,
    pub number: i32,
    pub status: String,
    pub position: String,
    #[serde(rename = "position_category")]
    pub position_category: String,
    pub height: String,
    pub weight: i32,
    pub age: String,
    pub college: String,
}
